src/main/java/mekanism/client/gui/element/button/ColorToggleButton.java

```diff
@@ -0,0 +1,52 @@
+package mekanism.client.gui.element.button;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import java.util.function.BooleanSupplier;
+import javax.annotation.Nonnull;
+import mekanism.api.text.EnumColor;
+import mekanism.client.gui.IGuiWrapper;
+import mekanism.client.gui.element.GuiElement;
+import mekanism.client.render.MekanismRenderer;
+import mekanism.common.lib.Color;
+import mekanism.common.tile.component.config.DataType;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+import org.jetbrains.annotations.Nullable;
+
+public class ColorToggleButton extends MekanismButton {
+
+    private final BooleanSupplier toggled;
+    private final EnumColor color;
+
+    public ColorToggleButton(IGuiWrapper gui, int x, int y, int size, BooleanSupplier toggled, Runnable onLeftClick, @Nullable GuiElement.IHoverable onHover) {
+        this(gui, x, y, size, EnumColor.DARK_BLUE, toggled, onLeftClick, onHover);
+    }
+
+    public ColorToggleButton(IGuiWrapper gui, int x, int y, int size, EnumColor color, BooleanSupplier toggled, Runnable onLeftClick, @Nullable GuiElement.IHoverable onHover) {
+        super(gui, x, y, size, size, TextComponent.EMPTY, onLeftClick, onHover);
+        this.toggled = toggled;
+        this.color = color;
+    }
+
+    @Override
+    public void drawBackground(@Nonnull PoseStack matrix, int mouseX, int mouseY, float partialTicks) {
+        if (toggled.getAsBoolean()) {
```

Comment:
This value should be stored in a local variable so as to not have to potentially calculate the supplier twice in one draw call if it has more logic than a simple true/false getter.